platform/ios/src/MGLMapView.mm

```diff
@@ -1610,19 +1608,16 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
             [MGLMapboxEvents ensureMetricsOptoutExists];
         }
     }
-    else if ([keyPath isEqualToString:@"coordinate"] && [object conformsToProtocol:@protocol(MGLAnnotation)])
+    else if ([keyPath isEqualToString:@"coordinate"] && [object conformsToProtocol:@protocol(MGLAnnotation)] && ![object isKindOfClass:[MGLMultiPoint class]])
     {
         id <MGLAnnotation> annotation = object;
-        MGLAnnotationTag annotationTag = [self annotationTagForAnnotation:annotation];
-        if (annotationTag != MGLAnnotationTagNotFound)
+        MGLAnnotationTag annotationTag = (MGLAnnotationTag)(NSUInteger)context;
```

Comment:
We do give MGLMultiPoints annotation tags, but we don’t bother to observe the `coordinate` key path on MGLMultiPoints, because MGLMultiPoints are immutable.

This `context` usage could potentially be unsafe if a developer subclasses MGLMapView and registers it as an observer for a non-multipoint annotation’s `coordinate` key path. The `context` can be used to distinguish multiple observations on the same key path and object, or to distinguish observation by multiple levels of the inheritance hierarchy, [but not both](https://mikeash.com/pyblog/key-value-observing-done-right.html).